tensorflow_addons/layers/tests/noisy_dense_test.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests NoisyDense layer."""
+
+
+import pytest
+import numpy as np
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow_addons.utils import test_utils
+from tensorflow_addons.layers.noisy_dense import NoisyDense
+from tensorflow.keras.mixed_precision.experimental import Policy
+
+
+def test_noisy_dense():
+    test_utils.layer_test(NoisyDense, kwargs={"units": 3}, input_shape=(3, 2))
+
+    test_utils.layer_test(NoisyDense, kwargs={"units": 3}, input_shape=(3, 4, 2))
+
+    test_utils.layer_test(NoisyDense, kwargs={"units": 3}, input_shape=(None, None, 2))
+
+    test_utils.layer_test(NoisyDense, kwargs={"units": 3}, input_shape=(3, 4, 5, 2))
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.parametrize("dtype", ["float16", "float32", "float64"])
+def test_noisy_dense_dtype(dtype):
+    inputs = tf.convert_to_tensor(
+        np.random.randint(low=0, high=7, size=(2, 2)), dtype=dtype
+    )
+    layer = NoisyDense(5, dtype=dtype)
+    outputs = layer(inputs)
+    np.testing.assert_array_equal(outputs.dtype, dtype)
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+def test_noisy_dense_with_policy():
```

Comment:
If I remember correctly we had a fixture like `@pytest.mark.usefixtures("run_with_mixed_precision_policy")` somewhere in the codebase

Reply:
Using the fixture would force the TensorSpec in the function to use a mixed policy when all I need is for the layer to do so.